File: AgentRoleJava/app/src/main/java/com/example/agentrolejava/ChatManager.java
package com.example.agentrolejava;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.CopyOnWriteArrayList;

public class ChatManager {
    private static ChatManager instance;
    private List<ChatMessage> messages;
    private ChatMessage tempMessage;
    private String agentStatus;
    private String agentStatusMessage;
    private List<ChatUpdateListener> listeners;

    public interface ChatUpdateListener {
        void onMessagesUpdated();
        void onTempMessageUpdated();
        void onAgentStatusUpdated();
    }

    private ChatManager() {
        messages = new ArrayList<>();
        listeners = new CopyOnWriteArrayList<>();
    }

    public static synchronized ChatManager getInstance() {
        if (instance == null) {
            instance = new ChatManager();
        }
        return instance;
    }

    public void addListener(ChatUpdateListener listener) {
        listeners.add(listener);
    }

    public void removeListener(ChatUpdateListener listener) {
        listeners.remove(listener);
    }

    /**
     * 添加消息
     */
    public void addMessage(String content, boolean isFromUser) {
        android.util.Log.d("ChatManager", "添加消息: " + content + ", 来自用户: " + isFromUser);
        // 清除临时消息
        tempMessage = null;
        // 添加正式消息
        messages.add(new ChatMessage(content, isFromUser));
        android.util.Log.d("ChatManager", "消息列表大小: " + messages.size());
        notifyMessagesUpdated();
        notifyTempMessageUpdated();
    }

    /**
     * 更新临时消息
     */
    public void updateTempMessage(String content, boolean isFromUser) {
        if (content != null && !content.isEmpty()) {
            tempMessage = new ChatMessage(content, isFromUser, true);
        } else {
            tempMessage = null;
        }
        notifyTempMessageUpdated();
    }

    /**
     * 更新Agent状态
     */
    public void updateAgentStatus(String status, String message) {
        this.agentStatus = status;
        this.agentStatusMessage = message;
        notifyAgentStatusUpdated();
    }

    /**
     * 清空消息列表
     */
    public void clearMessages() {
        messages.clear();
        tempMessage = null;
        agentStatus = null;
        agentStatusMessage = null;
        notifyMessagesUpdated();
        notifyTempMessageUpdated();
        notifyAgentStatusUpdated();
    }

    // Getters
    public List<ChatMessage> getMessages() {
        return new ArrayList<>(messages);
    }

    public ChatMessage getTempMessage() {
        return tempMessage;
    }

    public String getAgentStatus() {
        return agentStatus;
    }

    public String getAgentStatusMessage() {
        return agentStatusMessage;
    }

    // Notify methods
    private void notifyMessagesUpdated() {
        for (ChatUpdateListener listener : listeners) {
            listener.onMessagesUpdated();
        }
    }

    private void notifyTempMessageUpdated() {
        for (ChatUpdateListener listener : listeners) {
            listener.onTempMessageUpdated();
        }
    }

    private void notifyAgentStatusUpdated() {
        for (ChatUpdateListener listener : listeners) {
            listener.onAgentStatusUpdated();
        }
    }
} 

File: AgentRole/app/src/main/java/com/ainirobot/agent/sample/ui/components/ChatManager.kt
package com.ainirobot.agent.sample.ui.components

import androidx.compose.runtime.mutableStateListOf
import androidx.compose.runtime.mutableStateOf

/**
 * 聊天管理器，用于管理聊天消息
 */
object ChatManager {
    // 消息列表
    val messages = mutableStateListOf<ChatMessage>()
    
    // 临时消息，用于显示正在转写的文本
    val tempMessage = mutableStateOf<ChatMessage?>(null)

    // Agent状态
    val agentStatus = mutableStateOf<String?>(null)
    val agentStatusMessage = mutableStateOf<String?>(null)

    /**
     * 添加消息
     * @param content 消息内容
     * @param isFromUser 是否来自用户
     */
    fun addMessage(content: String, isFromUser: Boolean) {
        // 清除临时消息
        tempMessage.value = null
        // 添加正式消息
        messages.add(ChatMessage(content = content, isUser = isFromUser))
    }

    /**
     * 更新临时消息
     * @param content 临时消息内容
     * @param isFromUser 是否来自用户
     */
    fun updateTempMessage(content: String, isFromUser: Boolean) {
        if (content.isNotEmpty()) {
            tempMessage.value = ChatMessage(content = content, isUser = isFromUser)
        } else {
            tempMessage.value = null
        }
    }

    /**
     * 更新Agent状态
     * @param status 状态
     * @param message 状态消息
     */
    fun updateAgentStatus(status: String?, message: String?) {
        agentStatus.value = status
        agentStatusMessage.value = message
    }

    // 清空消息列表
    fun clearMessages() {
        messages.clear()
        tempMessage.value = null
        agentStatus.value = null
        agentStatusMessage.value = null
    }
} 